hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/PrngSystemPrecompiledContract.java

```diff
@@ -214,13 +214,8 @@ private void trackPrngOutput(
     }
 
     @VisibleForTesting
-    TransactionBody.Builder body(final Bytes randomNum) {
-        final var txnBody = TransactionBody.newBuilder();
-        if (randomNum == null) {
-            return txnBody;
-        }
-        final var body = UtilPrngTransactionBody.newBuilder();
-        return txnBody.setUtilPrng(body.build());
+    TransactionBody.Builder synthBody() {
+        return TransactionBody.newBuilder().setUtilPrng(UtilPrngTransactionBody.newBuilder());
```

Comment:
How does the random value get populated im the body?

Reply:
The random value bytes are populated in the child record of contractCall to `getPseudorandomSeed`.

Reply:
yeah, precompiles doesn't support giving a range parameter, since users can easily generate a pseudorandom number in a specified range using the seed by (integer value of seed % range) in solidity.
We just provide 256-bit pseudorandom seed using the first 256-bits of running hash of n-3 transaction record.